schemas_v2/thing.py

```diff
@@ -0,0 +1,220 @@
+# limitations under the License.
+# ===============================================================================
+from typing import List
+
+from pydantic import BaseModel, model_validator
+
+from schemas import ORMBaseModel
+from schemas_v2.location import LocationResponse
+
+
+# Thing
+class CreateThingIdLink(BaseModel):
+    """
+    Schema for creating a link between a thing and its ID.
+    """
+
+    thing_id: int
+    relation: str
+    alternate_id: str
+    alternate_organization: str
+
+
+class ThingResponse(ORMBaseModel):
+    name: str
+
+
+class UpdateThing(BaseModel):
+    """
+    Schema for updating a thing.
+    """
+
+    # location_id: int | None = None  # Optional location ID for the thing
+    name: str | None = None  # Optional name for the thing
+    # group: str | None = None  # Optional group for the thing
+    # description: str | None = None  # Optional description of the thing
+    # tags: list[str] | None = None  # Optional tags associated with the thing
+
+
+# Well
+class CreateWell(BaseModel):
+    """
+    Schema for creating a well.
+    """
+
+    location_id: int
+    name: str
+    # api_id: str | None = None
+    # ose_pod_id: str | None = None
+    well_type: str | None = None
+    well_depth: float | None = None  # in feet
+    hole_depth: float | None = None  # in feet
+    construction_notes: str | None = None
+
+    group: str | None = None  # Optional group for the well
+
+
+class CreateWellScreen(BaseModel):
+    """
+    Schema for creating a well screen.
+    """
+
+    well_id: int
+    screen_depth_bottom: float
+    screen_depth_top: float
+    screen_type: str | None = None
+
+    @model_validator(mode="after")
+    def validate_screen_type(self):
+        if self.screen_type is not None:
+            valid_screen_types = [
+                "PVC",
+            ]  # todo: get valid screen types from database
+            if self.screen_type not in valid_screen_types:
+                raise ValueError(
+                    f"Invalid screen_type: {self.screen_type}. "
+                    f"Valid options are: {', '.join(valid_screen_types)}."
+                )
+        return self
+
+    # validate that screen depth bottom is greater than top
+    @model_validator(mode="after")
+    def check_depths(self):
+        if self.screen_depth_bottom < self.screen_depth_top:
+            raise ValueError(
+                "screen_depth_bottom must be greater than screen_depth_top"
+            )
+        return self
+
+
+class WellResponse(ORMBaseModel):
+    """
+    Response schema for well details.
+    """
+
+    # api_id: str | None = None
+    # ose_pod_id: str | None = None
+    # usgs_id: str | None = None
+    well_type: str | None = None
+    well_depth: float | None = None  # in feet
+    hole_depth: float | None = None  # in feet
+    construction_notes: str | None = None
+    # Additional fields can be added as needed
+
+
+class LocationWellResponse(LocationResponse):
```

Comment:
Would the location be part of everything `Thing`'s `geometry` field? If so, `ThingReponse` can have a `geometry` field and then inheritance can be used to streamline fields across all things/types. Or, if we want to proceed with listing things at locations, that can be abstracted into a `LocationThingResponse` and then inheritance can be used for all thing types.